### ISSUE TITLE
KeyError when doing `conda-lock`

### ISSUE THREAD

**Issue by itamarst** [conda-incubator/conda-lock#67]
```shell-session
$ conda-lock -p linux-64
generating lockfile for linux-64

Traceback (most recent call last):
  File "/home/itamarst/.local/bin/conda-lock", line 8, in <module>
    sys.exit(main())
  File "/home/itamarst/.local/lib/python3.9/site-packages/conda_lock/conda_lock.py", line 400, in main
    run_lock(
  File "/home/itamarst/.local/lib/python3.9/site-packages/conda_lock/conda_lock.py", line 389, in run_lock
    make_lock_files(
  File "/home/itamarst/.local/lib/python3.9/site-packages/conda_lock/conda_lock.py", line 200, in make_lock_files
    lockfile_contents = create_lockfile_from_spec(
  File "/home/itamarst/.local/lib/python3.9/site-packages/conda_lock/conda_lock.py", line 251, in create_lockfile_from_spec
    url = fetch_by_dist_name[pkg["dist_name"]]["url"]
KeyError: '_libgcc_mutex-0.1-conda_forge'
```

Happens with both install via `pip` and install via `conda install`; should be latest version in both cases (0.7.2).

`environment.yml`:

```yaml
name: example
channels:
  - conda-forge
dependencies:
  - python=3.8
  - pandas=1.0
```

`.condarc`:

```yaml
channel_priority: strict
```

I also tried with Docker mode, but... that didn't seem to run a Docker container, to my confusion (#66) so can't tell if that'd fix it.

Deleting `.condarc` had no effect. `--no-mamba` had no effect. `-p mac-64` fails with a different package.

**Comment by itamarst** [comment]
Next, I tried doing it inside a 8-month-old `continuumio/miniconda3` Docker image I had lying around. Looks like it's the 4.8.2 one. That did work!

One difference:

1. The failure happened with Conda 4.9.1.
2. Success happened with Conda 4.8.2.

So in the miniconda container I upgraded to Conda 4.9.2... and it worked there too.

So at least now I have a workaround, but I'm still confused as why it's not working.
